src/loader/mod.rs

```diff
@@ -211,7 +211,7 @@ pub fn load_cmdline<M: GuestMemory>(
     guest_addr: GuestAddress,
     cmdline: &Cmdline,
 ) -> Result<()> {
-    let len = cmdline.as_str().len();
+    let len = cmdline.as_cstring().unwrap().to_bytes_with_nul().len();
```

Comment:
It's not safe to use unwraps in library code, you should either propagate the error or explain why the unwrap can not fail.
Also I don't understand the check below. If we use `to_bytes_with_nul` then we will have a nul terminator so the length can't be 0 anyway. I think we want to use `to_bytes` here.

Reply:
Yes, there is a problem here, but I would solve it by removing the unnecessary check if len is 0 (as you said, there is no possibility to have 0 bytes because we have the null terminator at least). In case of an empty cmdline, I would write the \0 to guest memory to make sure the kernel will stop parsing the cmdline right away - otherwise, we do not have any guarantee that there is a \0 where the kernel expect to find a cmdline.